helix-view/src/document.rs

```diff
@@ -1065,7 +1065,21 @@ impl Document {
             let write_result: anyhow::Result<_> = async {
                 let mut dst = tokio::fs::File::create(&write_path).await?;
                 to_writer(&mut dst, encoding_with_bom_info, &text).await?;
-                dst.sync_all().await?;
+                // Ignore ENOTSUP/EOPNOTSUPP (Operation not supported) errors from sync_all()
+                // This is known to occur on SMB filesystems on macOS where fsync is not supported
+                if let Err(e) = dst.sync_all().await {
+                    #[cfg(target_os = "macos")]
+                    {
+                        match e.raw_os_error() {
+                            Some(45) | Some(102) => {} // ENOTSUP or EOPNOTSUPP - ignore
```

Comment:
```suggestion
                            Some(libc::ENOTSUP) | Some(libc::EOPNOTSUPP) => ()
```

We can use `libc` here since we're within `target_os = "macos"` and libc is a dependency for Unix https://github.com/helix-editor/helix/blob/bfcbef10c513108c7b43317569416c2eefc4ed44/helix-view/Cargo.toml#L61-L62